type.h

```diff
@@ -81,6 +81,7 @@ typedef Typestruct* Datatype;
 
 //internal type functions
 int Simpletype_length(Simpletype s);
+int Type_extent(Datatype datatype, MPI_Aint * extent);
```

Comment:
Note this addition. This was needed for clang, and seems correct to do in that this function is referenced from other files besides type.c.